src/renderer/widgets/TextInput.tsx

```diff
@@ -26,45 +30,92 @@ const TextInput = forwardRef(
       handleChange,
       handleEscape,
       updateOnSubmitOnly,
+      validate,
+      formatInput = (s) => s,
     }: ITextInputProps,
     ref: ForwardedRef<HTMLInputElement>
```

Comment:
Do we ever use this ref in any component that contains a TextInput?

Reply:
Yeah we're using it in `PresetListItem` so we can identify when someone clicks outside of the text input.